ams/server/src/main/java/com/netease/arctic/server/utils/IcebergTableUtil.java

```diff
@@ -143,4 +157,43 @@ public static Set<DeleteFile> getDanglingDeleteFiles(Table internalTable) {
 
     return danglingDeleteFiles;
   }
+
+  /**
+   * Fetch all manifest files of an Iceberg Table
+   *
+   * @param table An iceberg table, or maybe base store or change store of mixed-iceberg format.
+   * @return Path set of all valid manifest files.
+   */
+  public static Set<String> getAllManifestFiles(Table table) {
+    Preconditions.checkArgument(
+        table instanceof HasTableOperations, "the table must support table operation.");
+    TableOperations ops = ((HasTableOperations) table).operations();
+
+    Table allManifest =
+        MetadataTableUtils.createMetadataTableInstance(
+            ops,
+            table.name(),
+            table.name() + "#" + MetadataTableType.ALL_MANIFESTS.name(),
+            MetadataTableType.ALL_MANIFESTS);
```

Comment:
This implementation does not fundamentally differ from the previous one, as it also internally calls the `snapshot.allManifests()` within this `ALL_MANIFESTS` Table.